engine/chain/internal/combined_txs_getter.go

```diff
@@ -0,0 +1,80 @@
+package internal
+
+import (
+	"context"
+	"maps"
+	"slices"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"iter"
+)
+
+// CombineTxsGetters creates a new CombinedTxsGetter
+func CombineTxsGetters(txsGetters ...chainmodels.TransactionsGetter) chainmodels.TransactionsGetter {
+	if len(txsGetters) == 0 {
+		return &emptyGetter{}
+	}
+	if len(txsGetters) == 1 {
```

Comment:
this should be rather after filterNilGetters

Reply:
I don't get your point.

Reply:
Aaa ok - changed 👍